### PR TITLE
Use `weights_only` only if torch >= 1.13

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

Fix https://github.com/huggingface/transformers/pull/27282#issuecomment-1887859328

cc @julien-c

### CODE REVIEW

**Comment by ydshieh** [comment]
failing tests are known flaky

**Review by LysandreJik — COMMENTED** [review]
Great! I think it can be simplified a little, but looks good otherwise

File: src/transformers/convert_pytorch_checkpoint_to_tf2.py [hunk 1]
```diff
@@ -329,7 +330,11 @@ def convert_pt_checkpoint_to_tf(
     if compare_with_pt_model:
         tfo = tf_model(tf_model.dummy_inputs, training=False)  # build the network
 
-        state_dict = torch.load(pytorch_checkpoint_path, map_location="cpu", weights_only=True)
+        state_dict = torch.load(
+            pytorch_checkpoint_path,
+            map_location="cpu",
+            weights_only=True if is_torch_greater_or_equal_than_1_13 else False,
```
[the remark this answers is not in the document]
Comment (ydshieh) [reply src/transformers/convert_pytorch_checkpoint_to_tf2.py]: thanks, very good suggestion 😅

**Review by LysandreJik — APPROVED** [review]
Thanks a lot @ydshieh!

**Pull request merged**